src/tests/relay/t_relay_io_silo.cpp

```diff
@@ -72,6 +72,34 @@ TEST(conduit_relay_io_silo, conduit_silo_cold_storage_generic_iface)
     EXPECT_EQ(n_load["c"].as_uint32(), c_val);
 }
 
+//-----------------------------------------------------------------------------
+// test silo file format detection
+TEST(conduit_relay_io_silo, test_silo_detect)
+{
+
+
+    // make sure bogus file doesn't return true
+    EXPECT_FALSE(io::silo::is_silo_file("BoGUS.txt"));
+    
+    // make sure pure hdf5 file doesn't return true
+    Node n_test;
+    n_test["a"] = 42.0;
+    io::save(n_test,"tout_hdf5_plain_vs_silo_open.hdf5");
+    EXPECT_FALSE(io::silo::is_silo_file("tout_hdf5_plain_vs_silo_open.hdf5"));
+
+    // make sure blueprint hdf5 file doesn't return true
+    n_test.reset();
+    blueprint::mesh::examples::braid("uniform", 5, 5, 0, n_test);
+    io::blueprint::save_mesh(n_test,"tout_bp_hdf5_mesh_vs_silo_open","hdf5");
+    EXPECT_FALSE(io::silo::is_silo_file("tout_bp_hdf5_mesh_vs_silo_open.root"));
+
+    std::string silo_input_file = utils::join_file_path("overlink", "box2d.silo");
+    silo_input_file = relay_test_silo_data_path(silo_input_file);
+    // make sure known silo file *does* return true
+    EXPECT_TRUE(io::silo::is_silo_file(silo_input_file));
```

Comment:
For completeness it might be good to have it try to open a Silo file written with the PDB driver.

Reply:
Do we have any examples in already in our repo?

Reply:
As far as I know, we do not. However, we can create them using the Silo Writer.

```
Node opts;
opts["silo_type"] = "pdb";
Node save_mesh;
blueprint::mesh::examples::basic("rectilinear", 3, 4, 0, save_mesh);
const std::string basename = "gimme_pdb";
io::silo::save_mesh(save_mesh, basename, opts);
```